### ISSUE TITLE
let_underscore_must_use can't be ignored

### ISSUE THREAD

**Issue by dylni** [rust-lang/rust-clippy#5366]
Clippy ignores `allow` annotations for [`let_underscore_must_use`](https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_must_use). However, it can sometimes be useful to suppress it.

main.rs:
```rust
#![warn(clippy::let_underscore_must_use)]

fn ok() -> Result<(), ()> {
    Ok(())
}

fn main() {
    #[allow(clippy::let_underscore_must_use)]
    let _ = ok();
}
```

```
$ cargo clippy
warning: non-binding let on an expression with `#[must_use]` type
 --> src/main.rs:9:5
  |
9 |     let _ = ok();
  |     ^^^^^^^^^^^^^
  |
note: lint level defined here
 --> src/main.rs:1:9
  |
1 | #![warn(clippy::let_underscore_must_use)]
  |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = help: consider explicitly using expression value
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_must_use
```

```
$ cargo clippy -V
clippy 0.0.212 (4ee1206 2020-02-01)
```

**Comment by flip1995** [comment]
This can be fixed by using [`check_local`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_lint/passes/trait.LateLintPass.html#method.check_local) instead of `check_stmt` here:

https://github.com/rust-lang/rust-clippy/blob/100a24d9d823e01483b3c2215dff71cadbf0ae50/clippy_lints/src/let_underscore.rs#L69

**Comment by pmk21** [comment]
Mind if I take up this issue?

**Comment by dylni** [comment]
@pmk21 That would be great. I'm not currently working on it.
